src/app/clusters/service-area-server/service-area-delegate.h

```diff
@@ -0,0 +1,424 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "service-area-cluster-objects.h"
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ServiceArea {
+
+class Instance;
+
+// *****************************************************************************
+// cluster constraints
+
+constexpr size_t kMaxNumSupportedLocations = 255;
+constexpr size_t kMaxNumSupportedMaps      = 255;
+constexpr size_t kMaxNumSelectedLocations  = 255;
+constexpr size_t kMaxNumProgressElements   = 255;
+
+constexpr size_t kMaxSizeStatusText = 256;
+
+/**
+ * Uncopyable is a helper to ensure that Service Area cluster instances and delegates
+ * are unique and cannot be copied.
+ */
+class Uncopyable
```

Comment:
This seems copy&pasted from operational-state-server.h

Why not just make the Delegate copy constructor and assignment operator `= delete`? Why the extra class if not shared code?

Reply:
Yes, from operational-state-server.h

Explicit class makes intent clearer, direct declaration is simpler. Style choice. I liked having the separate class, but no bit deal either way.

Reply:
Fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/078a3c909a79434b89e8167dd6e917b72b94cb6f).